src/cocoa/toga_cocoa/widgets/table.py

```diff
@@ -148,13 +170,38 @@ def change_source(self, source):
         self.table.reloadData()
 
     def insert(self, index, item):
-        self.table.reloadData()
+
+        # set parent = None if inserting to the root item
+        index_set = NSIndexSet.indexSetWithIndex(index)
+
+        self.table.insertRowsAtIndexes(
+            index_set,
+            withAnimation=NSTableViewAnimation.SlideDown
+        )
 
     def change(self, item):
-        self.table.reloadData()
+        row_index = self.interface.data.index(item)
```

Comment:
Even though we have access to the item index, would it make sense to use `_view_for_row` here? That would be guaranteed to be an O(1) lookup, whereas we don't know the complexity of `data.index()` (the default implementation is O(N)).

Reply:
Agreed. Since we already have the dict for lookup, we might as well make the most of it.